R/stop-level_obs_comments.R

```diff
@@ -1,121 +1,130 @@
+#       prior to 2022
 #------------------------------------------------------------------------------#
 
 #' Process species comments
 #' Takes data/mbbs.rda rows that have non NA species_comments
 #' and processes the row from a full-route summary of count for each species
 #' to break it out by stop_num.
-#' @param mbbs mbbs.rda
+#' @param ebird ebird data
 #' @importFrom dplyr group_by summarize filter anti_join select
 #' @importFrom tidyr pivot_longer
 #' @returns a df where each row is a count of a species at a stop
-process_species_comments <- function(mbbs) {
-  stopsmbbs <-
-    mbbs %>%
-    ungroup() %>%
-    prepare_to_process_sp_com()
-
-  # It would be great to modify this to use purrr:map at some point
-  # However, this short for loop also works nicely.
-  for (i in 1:nrow(stopsmbbs)) {
-    stopsmbbs[i, ] <- direct_to_sp_com_function(stopsmbbs[i, ])
-  }
-
-  # find rows where the output does not match count.
-  catch_errors <-
-    stopsmbbs %>%
-    group_by(sub_id, common_name, loc, year, species_comments) %>%
-    summarize(
-      out = sum(
-        s1, s2, s3, s4, s5, s6, s7, s8, s9, s10, s11, s12, s13, s14,
-        s15, s16, s17, s18, s19, s20
-      ),
-      count = count
-    ) %>%
-    filter(out != count)
-  # save error rows for assessment.
-  write.csv(catch_errors,
-    "inst/species_comments_errors.csv",
-    row.names = FALSE
-  )
-  # notify user of errors
-  cat(c(
-    "In processing species_comments:",
-    "\n",
-    nrow(catch_errors),
-    "rows have been removed where the sum of the processed counts",
-    "did not match the original ebird checklist's route-total count",
-    "for that species",
-    "\n",
-    "To follow up see:",
-    "inst/species_comments_errors.csv"
-  ))
-
-  # remove rows where output doesn't match count
-  stopsmbbs <- anti_join(stopsmbbs, catch_errors)
-
-  # pivot s1:s20 into stop_num and give count for each stop
-  stopsmbbs <-
-    stopsmbbs %>%
-    select(-stop_num, -count, -sc_note) %>%
-    pivot_longer(
-      cols = s1:s20,
-      names_to = "stop_num", names_prefix = "s", values_to = "count"
-    )
-
-  stopsmbbs # returns
+process_obs_details <- function(ebird) {
+  obs <-
+    ebird |>
+    ungroup() |>
+    prep_obs_details_data()
+
+  obs |>
+   mutate(
+    stop_data = 
+        stringr::str_split(obs$obs_details, ",|;") |>
+        purrr::map2(
+          .y = obs$count,
+          ~ { str_replace_all(.x, c("stop" = "", "st" = "", "\\(\\)" = "")) |>
+              trimws() |>
+              pad_or_truncate() |>
+              (\(x) {
+                tibble(
+                  stop_num = 1:20,
+                  count  = `if`(
+                              length(x) == 20,
+                              parse_lengtheq20(x),
+                              parse_lengthlt20(x, .y)
+                            )
+                )
+              })()
+          })
+   )
+
+
+  
+  # # find rows where the output does not match count.
```

Comment:
Add back error catching/handling. I've been thinking that we want to add a logging functionaliity to our build process because there are multiple places in the build where we want to identify possible problems.